spec/services/teacher_training_public_api/sync_provider_spec.rb

```diff
@@ -6,42 +6,381 @@
   describe '.call' do
     context 'ingesting a brand new provider' do
       it 'just creates the provider without any courses' do
-        provider_from_api = fake_api_provider(code: 'ABC')
-
-        described_class.new(
-          provider_from_api: provider_from_api,
-          recruitment_cycle_year: RecruitmentCycle.current_year,
-        ).call
-
-        provider = Provider.find_by_code('ABC')
+        provider_from_api = fake_api_provider({ code: 'ABC' })
+        described_class.new(provider_from_api: provider_from_api, recruitment_cycle_year: stubbed_recruitment_cycle_year).call
 
+        provider = Provider.find_by(code: 'ABC')
         expect(provider).to be_present
         expect(provider.courses).to be_blank
       end
     end
 
-    context 'ingesting a provider configured to sync courses' do
-      it 'assigns qualifications and program_type to an existing course' do
-        provider = create :provider, code: 'ABC', sync_courses: true
-        course = create(:course, code: 'ABC1', provider: provider)
+    context 'ingesting an existing provider not configured to sync courses' do
```

Comment:
It kind of feels that this should test that it calls the sync course job with the correct parameters and all of these tests should sit in the sync courses spec.

What do you think?

Reply:
Good idea, I've refactored that now. But I'm struggling a little with writing the test for checking that `sync_courses` is called

Reply:
I've just pushed what I've got so far, but it'd be good to chat about what I'm missing